tests/test_phase0.py

```
@@ -0,0 +1,100 @@
+            text=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        self.assertEqual(42, result.returncode)
+        self.assertIn("intentional phase-0 delivery failure control", result.stderr)
+
+    def test_ship_rejects_an_unsafe_checkout(self) -> None:
+        marker = ROOT / ".phase0-ship-test-marker"
+        marker.write_text("negative control\n", encoding="utf-8")
+        try:
+            result = subprocess.run(
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)